File: backend/src/main/java/com/construction/client/controller/inbox_message.dart
class InboxMessage {
  final int nid;
  final String accountId;
  final String? category;
  final String title;
  final String body;
  final int? targetId;
  bool isRead;
  final DateTime createdAt;

  InboxMessage({
    required this.nid,
    required this.accountId,
    this.category,
    required this.title,
    required this.body,
    this.targetId,
    required this.isRead,
    required this.createdAt,
  });

  factory InboxMessage.fromJson(Map<String, dynamic> json) {
    return InboxMessage(
      nid: json['nid'],
      accountId: json['accountid'],
      category: json['category'],
      title: json['title'],
      body: json['body'],
      targetId: json['target_id'],
      isRead: json['isRead'] ?? false,
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}

File: backend/src/main/java/com/construction/client/entity/InboxMessage.java
package com.construction.client.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "salinbox_messages")
public class InboxMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "nid")
    private Long nid;

    @Column(name = "accountid")
    private String accountId;

    @Column(name = "fcm_token")
    private String fcmToken;

    @Column(name = "category")
    private String category;

    @Column(name = "title")
    private String title;

    @Column(name = "body", columnDefinition = "TEXT")
    private String body;

    @Column(name = "target_id")
    private Integer targetId;

    @Column(name = "is_read")
    private Boolean isRead = false;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    // Getters and Setters
    public Long getNid() {
        return nid;
    }

    public void setNid(Long nid) {
        this.nid = nid;
    }

    public String getAccountId() {
        return accountId;
    }

    public void setAccountId(String accountId) {
        this.accountId = accountId;
    }

    public String getFcmToken() {
        return fcmToken;
    }

    public void setFcmToken(String fcmToken) {
        this.fcmToken = fcmToken;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public Integer getTargetId() {
        return targetId;
    }

    public void setTargetId(Integer targetId) {
        this.targetId = targetId;
    }

    public Boolean getIsRead() {
        return isRead;
    }

    public void setIsRead(Boolean isRead) {
        this.isRead = isRead;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}